service/controller/resource/workermigration/create.go

```diff
@@ -0,0 +1,505 @@
+	o := client.MatchingLabels{
+		capiv1alpha3.ClusterLabelName: key.ClusterID(&cr),
+	}
+
+	var dcList corev1alpha1.DrainerConfigList
+	err := r.ctrlClient.List(ctx, &dcList, o)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	for _, dc := range dcList.Items {
+		if dc.Status.HasTimeoutCondition() {
+			err = r.ctrlClient.Delete(ctx, &dc)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func (r *Resource) deleteDrainerConfigs(ctx context.Context, cr providerv1alpha1.AzureConfig) error {
+	err := r.ctrlClient.DeleteAllOf(ctx, &corev1alpha1.DrainerConfig{}, client.MatchingLabels{capiv1alpha3.ClusterLabelName: key.ClusterID(&cr)}, client.InNamespace(key.ClusterID(&cr)))
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (r *Resource) ensureAzureMachinePoolExists(ctx context.Context, cr providerv1alpha1.AzureConfig, vmSize string) (expcapzv1alpha3.AzureMachinePool, error) {
+	var azureMachinePool expcapzv1alpha3.AzureMachinePool
+
+	nsName := types.NamespacedName{
+		Name:      cr.GetName(),
+		Namespace: key.OrganizationNamespace(&cr),
+	}
+	err := r.ctrlClient.Get(ctx, nsName, &azureMachinePool)
+	if err == nil {
+		// NOTE: Success return when AzureMachinePool CR already exists.
+		return azureMachinePool, nil
+	} else if errors.IsNotFound(err) {
+		// This is ok. CR gets created in a bit.
+	} else if err != nil {
+		return expcapzv1alpha3.AzureMachinePool{}, microerror.Mask(err)
+	}
+
+	var dockerDiskSizeGB int
+	var kubeletDiskSizeGB int
+	{
+		if key.WorkerCount(cr) > 0 {
+			dockerDiskSizeGB = cr.Spec.Azure.Workers[0].DockerVolumeSizeGB
+			kubeletDiskSizeGB = cr.Spec.Azure.Workers[0].KubeletVolumeSizeGB
+		}
+
+		if dockerDiskSizeGB <= 0 {
+			dockerDiskSizeGB = 100
+		}
+
+		if kubeletDiskSizeGB <= 0 {
+			kubeletDiskSizeGB = 100
+		}
+	}
+
+	// CR didn't exist so it's created here.
+	azureMachinePool = expcapzv1alpha3.AzureMachinePool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name,
+			Namespace: key.OrganizationNamespace(&cr),
+			Labels: map[string]string{
+				label.AzureOperatorVersion:    key.OperatorVersion(&cr),
+				label.Cluster:                 key.ClusterID(&cr),
+				capiv1alpha3.ClusterLabelName: key.ClusterName(&cr),
+				label.MachinePool:             key.ClusterID(&cr),
+				label.Organization:            key.OrganizationID(&cr),
+				label.ReleaseVersion:          key.ReleaseVersion(&cr),
+			},
+		},
+		Spec: expcapzv1alpha3.AzureMachinePoolSpec{
+			Location: r.location,
+			Template: expcapzv1alpha3.AzureMachineTemplate{
+				DataDisks: []capzv1alpha3.DataDisk{
+					{
+						NameSuffix: "docker",
+						DiskSizeGB: int32(dockerDiskSizeGB),
+						Lun:        to.Int32Ptr(21),
+					},
+					{
+						NameSuffix: "kubelet",
+						DiskSizeGB: int32(kubeletDiskSizeGB),
+						Lun:        to.Int32Ptr(22),
+					},
+				},
+				VMSize: vmSize,
+			},
+		},
+	}
+
+	err = r.ctrlClient.Create(ctx, &azureMachinePool)
```

Comment:
wasn't the `capzcrs` handler creating the `MachinePool` and `AzureMachinePool`?

Reply:
`capzcrs` takes care of the CP CRs. `AzureConfig` -> {`Cluster`, `AzureCluster`, `AzureMachine`}.